src/Mpdf.php

```diff
@@ -10458,7 +10459,11 @@ function watermark($texte, $angle = 45, $fontsize = 96, $alpha = 0.2)
 			$this->watermark_font = $this->default_font;
 		}
 
-		$this->SetFont($this->watermark_font, "B", $fontsize, false); // Don't output
+		if (!$this->watermark_font_size) {
+			$this->watermark_font_size = $this->default_font_size;
```

Comment:
Changing a config setting internally is a shady technique an will lead to unexpected results. 

Use the method parameter (see below) and optionally validate its value.